### PR TITLE
Added function parameter definitions to funcsrules

### PR DESCRIPTION
**Pull request by situx**
I added function parameter and return value definitions to the funcsrules vocabulary, only for functions.
I used the FNO vocabulary: https://fno.io/spec/#dfn-function-declaration

### CODE REVIEW

**Review by nicholascar — APPROVED** [review]

File: 1.0/funcsrules.ttl [hunk 1]
```diff
@@ -160,86 +162,171 @@ rules:
...
+    fno:required "true"^^xsd:boolean ;
+    skos:prefLabel "geom"@en .
+
+funcs:distance_param2 a fno:Parameter ;
+    fno:type ogc:wktLiteral, ogc:gmlLiteral  ;
+    fno:required "true"^^xsd:boolean ;
+    skos:prefLabel "geom2"@en .
+
+funcs:distance_param3 a fno:Parameter ;
+    fno:type ogc:wktLiteral, ogc:gmlLiteral  ;
```
Comment (mperry455) [inline 1.0/funcsrules.ttl:311]: Should be xsd:anyURI

File: 1.0/funcsrules.ttl [hunk 2]
```diff
@@ -549,29 +919,66 @@ DE-9IM: TTTFTTFFT"""@en ;
...
+
+funcs:relate_param2 a fno:Parameter ;
+    fno:type ogc:wktLiteral, ogc:gmlLiteral  ;
+    fno:required "true"^^xsd:boolean ;
+    skos:prefLabel "geom2"@en .
+
+funcs:relate_param3 a fno:Parameter ;
+    fno:type xsd:string ;
+    fno:required "true"^^xsd:boolean ;
+    skos:prefLabel "geom2"@en .
```
Comment (mperry455) [inline 1.0/funcsrules.ttl:966]: Should be "pattern-matrix"@en

File: 1.0/funcsrules.ttl [hunk 3]
```diff
@@ -698,36 +1219,77 @@ DE-9IM: T*F**F***"""@en ;
...
     skos:definition "A query function that returns a geometry consisting of all points that are part of only one input geometry."@en ;
     skos:prefLabel "symmetric difference"@en ;
 .
 
+funcs:symDifference_param1 a fno:Parameter ;
+    fno:type ogc:wktLiteral, ogc:gmlLiteral  ;
+    fno:required "true"^^xsd:boolean ;
+    skos:prefLabel "geom1"@en .
+
+funcs:sfWithin_output a fno:Output ;
```
Comment (mperry455) [inline 1.0/funcsrules.ttl:1258]: Should be funcs:symDifference_outpue

File: 1.0/funcsrules.ttl [hunk 4]
```diff
@@ -698,36 +1219,77 @@ DE-9IM: T*F**F***"""@en ;
...
 funcs:symDifference
     a
         skos:Concept ,
-        sd:Function ;
+        sd:Function, fno:Function ;
     dcterms:contributor "Matthew Perry" ;
     dcterms:creator "OGC GeoSPARQL 1.0 Standard Working Group" ;
     dcterms:date "2011-06-16"^^xsd:date ;
     dcterms:source <http://www.opengis.net/doc/geosparql/1.0> ;
+    fno:expects (funcs:symDifference_param1 ) ;
```
Comment (mperry455) [inline 1.0/funcsrules.ttl:1244]: We need a geometry param2

File: 1.1/funcsrules.ttl [hunk 5]
```diff
@@ -577,61 +777,154 @@ funcs:boundingCircle
...
+    fno:required "true"^^xsd:boolean ;
+    skos:prefLabel "geom"@en .
+
+funcs:distance_param2 a fno:Parameter ;
+    fno:type geo:wktLiteral, geo:gmlLiteral, geo:geoJSONLiteral, geo:kmlLiteral, geo:dggsLiteral  ;
+    fno:required "true"^^xsd:boolean ;
+    skos:prefLabel "geom2"@en .
+
+funcs:distance_param3 a fno:Parameter ;
+    fno:type geo:wktLiteral, geo:gmlLiteral, geo:geoJSONLiteral, geo:kmlLiteral, geo:dggsLiteral  ;
```
Comment (mperry455) [inline 1.1/funcsrules.ttl:914]: Should be xsd:anyURI

File: 1.1/funcsrules.ttl [hunk 6]
```diff
@@ -936,20 +1596,58 @@ DE-9IM: TTTFTTFFT"""@en ;
...
+
+funcs:relate_param2 a fno:Parameter ;
+    fno:type geo:wktLiteral, geo:gmlLiteral, geo:geoJSONLiteral, geo:kmlLiteral, geo:dggsLiteral  ;
+    fno:required "true"^^xsd:boolean ;
+    skos:prefLabel "geom2"@en .
+
+funcs:relate_param3 a fno:Parameter ;
+    fno:type xsd:string ;
+    fno:required "true"^^xsd:boolean ;
+    skos:prefLabel "geom2"@en .
```
Comment (mperry455) [inline 1.1/funcsrules.ttl:1639]: Should be "pattern-matrix"@en

File: 1.1/funcsrules.ttl [hunk 7]
```diff
@@ -1045,27 +1855,68 @@ DE-9IM: T*F**F***"""@en ;
...
     skos:definition "A query function that returns a geometry consisting of all points that are part of only one input geometry."@en ;
     skos:prefLabel "symmetric difference"@en ;
 .
 
+funcs:symDifference_param1 a fno:Parameter ;
+    fno:type geo:wktLiteral, geo:gmlLiteral, geo:geoJSONLiteral, geo:kmlLiteral, geo:dggsLiteral  ;
+    fno:required "true"^^xsd:boolean ;
+    skos:prefLabel "geom1"@en .
+
+funcs:sfWithin_output a fno:Output ;
```
Comment (mperry455) [inline 1.1/funcsrules.ttl:1889]: Should be funcs:symDifference_output

File: 1.1/funcsrules.ttl [hunk 8]
```diff
@@ -1045,27 +1855,68 @@ DE-9IM: T*F**F***"""@en ;
...
+    fno:required "true"^^xsd:boolean ;    
+    fno:type xsd:boolean  .
+
 funcs:symDifference
-    a sd:Function ;
+    a sd:Function, fno:Function ;
     dcterms:source <http://www.opengis.net/doc/geosparql/1.1> ;
     policy:status status:valid ;
+    fno:expects (funcs:symDifference_param1 ) ;
```
Comment (mperry455) [inline 1.1/funcsrules.ttl:1876]: We need a geom2 parameter

**Review by mperry455 — APPROVED** [review]
Approved but the minor errors noted in comments need to be fixed.

**Comment by situx** [comment]
All done, merging now!

**Pull request merged**